libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m

```diff
@@ -184,10 +187,19 @@ - (id) initWithConfig:(SFHybridViewConfig *) viewConfig useUIWebView:(BOOL) useU
         _hybridViewConfig = (viewConfig == nil ? [SFHybridViewConfig fromDefaultConfigFile] : viewConfig);
         NSAssert(_hybridViewConfig != nil, @"_hybridViewConfig was not properly initialized. See output log for errors.");
         self.startPage = _hybridViewConfig.startPage;
-        
+
         // Setup global stores and syncs defined in static configs
         [[SalesforceHybridSDKManager sharedManager] setupGlobalStoreFromDefaultConfig];
         [[SalesforceHybridSDKManager sharedManager] setupGlobalSyncsFromDefaultConfig];
+        __weak typeof(self) weakSelf = self;
+        [SFSDKAuthConfigUtil getMyDomainAuthConfig:^(SFOAuthOrgAuthConfiguration *authConfig, NSError *error) {
```

Comment:
(Sorry, late to the party on this review.)

In general, `init` methods are not a great place to take actions—particularly network activities.  And even more particularly, asynchronous network activities—init methods should be synchronous.

This is really looking for a different design pattern.  Some suggestions:

1. Given that this information is general, and likely to become germane to more parts of the SDK footprint than the hybrid view, shouldn't this happen as part of the SDK bootstrapping process?
2. If this is going to stay in the hybrid view controller, it really needs to move somewhere that's part of an existing async flow.  I would suggest `viewDidLoad`.

Reply:
Another point, probably in favor of the first approach: this information feels pretty tightly coupled to the authentication of a user.  Shouldn't that process be the process that owns the retrieval and management of this configuration?

Reply:
A further point that puts me 99% in favor of the first approach: we're [already retrieving this data](https://github.com/forcedotcom/SalesforceMobileSDK-iOS/blob/dev/libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m#L230) as part of authentication, which is making this an additional, redundant network call.  We should simply persist the retrieved data as part of e.g. the user account, and reference it here.

Reply:
Thanks for the feedback, @khawkins. Yes - we are already retrieving the data. I'll look into a persistence solution.